public/app/features/alerting/unified/rule-list/hooks/prometheusGroupsGenerator.ts

```diff
@@ -95,6 +96,23 @@ export function useGrafanaGroupsGenerator(hookOptions: UseGeneratorHookOptions =
   );
 }
 
+/**
+ * Converts a Prometheus groups generator yielding arrays of groups to a generator yielding groups one by one
+ * @param generator - The paginated generator to convert
+ * @returns A non-paginated generator that yields all groups from the original generator one by one
+ */
+export function toPageless<TGroup extends PromRuleGroupDTO>(
```

Comment:
nit: I was initially a bit confused about what this function did from the consuming side; perhaps something like `yieldIndividualRuleGroups` would be a bit more verbose but clearer when reading?